cpp/src/gandiva/gdv_function_stubs_test.cc

```diff
@@ -766,4 +766,40 @@ TEST(TestGdvFnStubs, TestCastVarbinaryFloat8) {
   ctx.Reset();
 }
 
+TEST(TestGdvFnStubs, TestMaskFirstN) {
```

Comment:
Did you check what Dremio does if mask_first_n() is passed a negative number as an argument for the number of digits to mask? Does the planner handle that? or should the implementation of the function handle this case?

Reply:
If it passes a negative number, it will threat it as a positive number, so I did the handling of the negative number on the function